### PR TITLE
$osu -recent

### PR DESCRIPTION
**Pull request by hburn7**

### CODE REVIEW

File: Kaguya/Kaguya/Discord/Commands/Reference/Osu.cs [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+        [Priority(1)]
+        [Command("-set")]
+        [Summary("Give me an osu! username to remember you by! After setting a username, you will no longer have " +
+                 "to specify a username when using osu commands. You must also configure your game mode.\n\n" + GAME_MODE_STRING)]
+        [Remarks("<game mode> <username or id>")]
+        [Example("std name with spaces")]
+        [Example("mania 99999999")]
+        [Example("ctb really good ctb player")]
+        public async Task OsuSetCommand(string gameModeString, [Remainder] string username)
```
Comment (BitMasher) [inline Kaguya/Kaguya/Discord/Commands/Reference/Osu.cs:42]: should gameModeString have input validation? there is no input validation in this method

File: Kaguya/Kaguya/Discord/Commands/Reference/Osu.cs [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        }
+
+        [Priority(2)]
+        [Command("-recent", RunMode = RunMode.Async)]
+        [Alias("-r")]
+        [Summary("Displays the most recent play for a given username. The `gamemode` parameter is only " +
+                 "optional if your username has been set via the `osu -set` command.\n\n" + GAME_MODE_STRING)]
+        [Remarks("\n<gamemode> [username]")]
+        // ReSharper disable once MethodOverloadWithOptionalParameter
+        public async Task OsuRecentCommand(string gameMode = null, [Remainder] string username = null)
```
Comment (BitMasher) [inline Kaguya/Kaguya/Discord/Commands/Reference/Osu.cs:71]: gameMode has no input validation, should this be validated?

File: Kaguya/Kaguya/Discord/DiscordExtensions/TimeExtensions.cs [hunk 3]
```diff
@@ -11,5 +12,13 @@ public static string HumanizeTraditionalReadable(this TimeSpan t)
         {
             return t.Humanize(3, minUnit: TimeUnit.Second, maxUnit: TimeUnit.Day);
         }
+
+        public static DateTime GetCurrentDaylightSavingsUtcTime()
```
Comment (BitMasher) [inline Kaguya/Kaguya/Discord/DiscordExtensions/TimeExtensions.cs]: just use DateTime.UtcNow: https://docs.microsoft.com/en-us/dotnet/api/system.datetime.utcnow?view=net-5.0

File: Kaguya/Kaguya/External/Osu/OsuData.cs [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        
+        /// <summary>
+        /// std = Standard
+        /// mania = Mania
+        /// taiko = Taiko
+        /// ctb = Catch
+        /// </summary>
+        /// <param name="gmString"></param>
+        /// <returns></returns>
+        public GameMode GetGamemode(string gmString)
```
Comment (BitMasher) [inline Kaguya/Kaguya/External/Osu/OsuData.cs]: Following naming conventions this should be GetGameMode

**Review by BitMasher — CHANGES_REQUESTED** [review]

File: Kaguya/Kaguya/Discord/Commands/Reference/Osu.cs (hunk 2 above)
Comment (hburn7) [reply Kaguya/Kaguya/Discord/Commands/Reference/Osu.cs:71]: Validation is done in the OsuData.cs `GetGameMode()` function.

File: Kaguya/Kaguya/Discord/DiscordExtensions/TimeExtensions.cs (hunk 3 above)
Comment (hburn7) [reply Kaguya/Kaguya/Discord/DiscordExtensions/TimeExtensions.cs]: Yeah..I figured out the proper implementation of that actually. Thought something was messed up where my `DateTime.UtcNow` wasn't working, but the machine was right of course. Turns out this method isn't even used anywhere else.

File: Kaguya/Kaguya/External/Osu/OsuRecent.cs [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+    {
+        private readonly OsuData _osuData;
+        private readonly ICommandContext _ctx;
+
+        private const char POINT_RIGHT = '▸';
+
+        public OsuRecent(OsuData osuData, ICommandContext ctx) : base(osuData.Client)
+        {
+            _osuData = osuData;
+            _ctx = ctx;
```
Comment (hburn7) [inline Kaguya/Kaguya/External/Osu/OsuRecent.cs]: _ctx isn't used anywhere else in the file.

**Pull request merged**